c/misra/src/rules/RULE-6-1/BitFieldsShallOnlyBeDeclaredWithAnAppropriateType.ql

```diff
@@ -12,22 +12,33 @@
 
 import cpp
 import codingstandards.c.misra
+import codingstandards.cpp.Compiler
 
-predicate isAppropriatePrimitive(Type type) {
-  /* An appropriate primitive types to which a bit-field can be declared. */
-  type instanceof IntType and
+Type getSupportedBitFieldType(Compiler compiler) {
+  compiler instanceof UnsupportedCompiler and
   (
-    type.(IntegralType).isExplicitlySigned() or
-    type.(IntegralType).isExplicitlyUnsigned()
+    result instanceof IntType and
+    (
+      result.(IntegralType).isExplicitlySigned() or
+      result.(IntegralType).isExplicitlyUnsigned()
+    )
+    or
+    result instanceof BoolType
   )
   or
-  type instanceof BoolType
+  (compiler instanceof Gcc or compiler instanceof Clang) and
```

Comment:
just a quick check, where did you find that clang should be the same as gcc (vs just having support for unsigned long and unsigned long long, as described in the [open std changenote](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2774.htm)?)

Reply:
Clang is suppose to be a drop in replacement for Gcc. Based on testing of the version we support I determined that they support the same bitmask types. See https://github.com/github/codeql-coding-standards/issues/318#issuecomment-2021129560 for the details of that test.

Reply:
ah sorry missed that, that looks good to me!